dcm4che-conf/dcm4che-conf-dicom/src/test/java/org/dcm4che3/conf/dicom/DicomConfigurationTest.java

```diff
@@ -91,6 +93,88 @@ public void renameAETest() throws ConfigurationException {
 
     }
 
+    @Test
+    public void listDevices_produce_expected_results() {
+        CommonDicomConfiguration config = SimpleStorageTest.createCommonDicomConfiguration();
+        config.purgeConfiguration();
+
+        Device originalDevice = createDeviceWithArchiveExtension("DEVICE", "Default", "Other", "Storage", DeviceType.ARCHIVE, DeviceType.PRINT);
+        config.persist(originalDevice);
+
+        Device[] devices = config.listDevices(DeviceType.ARCHIVE);
+        Assert.assertTrue(devices.length == 1);
+
+        Device device = devices[0];
+
+        Assert.assertTrue(deepCompare(originalDevice, device));
+        Assert.assertTrue(Arrays.asList(device.getPrimaryDeviceTypes()).contains(DeviceType.ARCHIVE.toString()));
+        Assert.assertTrue(Arrays.asList(device.getPrimaryDeviceTypes()).contains(DeviceType.PRINT.toString()));
+        Assert.assertTrue(device.getApplicationEntities().stream().anyMatch(aet -> {
+            ExternalArchiveAEExtension extension = aet.getAEExtension(ExternalArchiveAEExtension.class);
+            return extension != null && extension.isDefaultForStorage();
+        }));
+    }
+
+    @Test
+    public void listDevices_produce_equivalent_results_as_findDevice() {
+        CommonDicomConfiguration config = SimpleStorageTest.createCommonDicomConfiguration();
+        config.purgeConfiguration();
+
+        Device originalDevice = createDeviceWithArchiveExtension("DEVICE", "Default", "Other", "Storage", DeviceType.ARCHIVE, DeviceType.PRINT);
+        config.persist(originalDevice);
+
+        Device[] devices = config.listDevices(DeviceType.ARCHIVE);
+        Assert.assertTrue(devices.length == 1);
+
+        Device listedDevice = devices[0];
+        Device foundDevice = config.findDevice(originalDevice.getDeviceName());
+
+        Assert.assertTrue(deepCompare(foundDevice, listedDevice));
+    }
+
+    @Ignore
```

Comment:
Why is this on `@Ignore`? I'd just remove it if you're not going to run it.

Reply:
I initially thought about keeping this test method as a reference. But I'll remove it since it doesn't seem to provide any value.
I'll also improve the assertion messages for the other methods I had created

Reply:
Pushed 8bccf310095393fbf90ab1291b00c381d251554f to address changes on DicomConfigurationTest